lib/jsonapi/deserializable/resource.rb

```diff
@@ -1,110 +1,155 @@
-require 'jsonapi/deserializable/resource_dsl'
+require 'jsonapi/deserializable/resource/configuration'
+require 'jsonapi/deserializable/resource/dsl'
+require 'jsonapi/parser/resource'
 
 module JSONAPI
   module Deserializable
     class Resource
-      include ResourceDSL
+      extend DSL
 
       class << self
         attr_accessor :type_block, :id_block, :attr_blocks,
-                      :has_one_rel_blocks, :has_many_rel_blocks
+                      :has_one_rel_blocks, :has_many_rel_blocks,
+                      :configuration
       end
 
-      self.attr_blocks = {}
+      @class_cache = {}
```

Comment:
I guess it won't matter too much, but keep in mind this means a separate cache for every class inheriting this. Something like `JSONAPI::Deserializable::ClassCache` or something similar would avoid this.